client/webserver/site/src/js/markets.js

```diff
@@ -1464,23 +1482,28 @@ export default class MarketsPage extends BasePage {
     // Handle market order differently.
     if (order.rate === 0) {
       // This is a market order.
-      if (!row || row.order.rate !== 0) {
-        row = this.orderTableRow(order, cssClass)
+      if (row && row.getRate() === 0) {
+        row.insertOrder(order)
+      } else {
+        row = this.orderTableRow([order], cssClass)
         tbody.insertBefore(row, tbody.firstChild)
       }
-      row.addQty(order.qty)
       return
     }
     // Must be a limit order. Sort by rate. Skip the market order row.
-    if (row && row.order.rate === 0) row = row.nextSibling
-    const tr = this.orderTableRow(order, cssClass)
+    if (row && row.getRate() === 0) row = row.nextSibling
     while (row) {
-      if ((order.rate < row.order.rate) === order.sell) {
+      if (order.rate === row.getRate()) {
+        row.insertOrder(order)
+        return
+      } else if ((order.rate < row.getRate()) === order.sell) {
+        const tr = this.orderTableRow([order], cssClass)
         tbody.insertBefore(tr, row)
         return
       }
       row = row.nextSibling
     }
```

Comment:
You'll need to figure out a way to separate epoch queue orders and booked orders and create a row for each.